frontend/src/api/assignments.ts

```diff
@@ -56,48 +56,41 @@ export const createAssignmentTemplate = async (
       headers: {
         "Content-Type": "application/json",
       },
-      body: JSON.stringify({
-        template_repo_id: assignment.id,
-        template_repo_owner: assignment.owner,
-      })
+      body: JSON.stringify(template),
```

Comment:
trusting the frontend to pass a valid template is bad design. It should pass only the identifying information for the template imo